pkg/providers/tinkerbell/reconciler/reconciler.go

```diff
@@ -59,10 +59,29 @@ func (r *Reconciler) Reconcile(ctx context.Context, log logr.Logger, cluster *an
 
 	return controller.NewPhaseRunner().Register(
 		r.ipValidator.ValidateControlPlaneIP,
+		r.ValidateClusterSpec,
+		r.ReconcileControlPlane,
```

Comment:
I added this to ensure the flow of the phase runner is clear, ie, we check validations, then reconcile CP before reconcile CNI. Since reconcile CP isn't implemented yet, i added this to the Phase runner without causing any consequences.